### PR TITLE
CVL Changes #1: Initial CVL code reorganization and common infra changes

### PR DESCRIPTION
**Pull request by dutta-partha**
**- Why I did it**

As part of CVL code update and new features addition, the existing CVL code has to be reorganized and common infra needs to be changed.

**- How I did it**

N/A. 
Please refer to change log for more details.

**- How to verify it**

No specific test cases to be executed.  Run cvl go test cases using - 'make cvl-test'.

**- Description for the changelog**

1. Modify Makefiles
   - Modify CVL Makefile for cleanall rule
   - Modify Makefile for CVL YIN schema generation
   - Modify Makefile for CVL test YIN schema generation
2. Moved cache, syntax and semantic related code from cvl.go to cvl_cache.go, cvl_syntax.go and cvl_semantics.go respectively .
3. Added new API signatures for integration. Implementation will be added in subsequent PRs.
4. Copied sonic-port.yang and sonic-acl.yang to CVL own's test folder.

### CODE REVIEW

File: cvl/cvl.go [hunk 1]
```diff
@@ -25,15 +25,15 @@ import (
 	"regexp"
 	"time"
 	 log "github.com/golang/glog"
-	"encoding/json"
+	//"encoding/json"
```
Comment (renukamanavalan) [inline cvl/cvl.go:28]: Please remove, if not required

File: cvl/cvl.go [hunk 2]
```diff
@@ -25,15 +25,15 @@ import (
...
 	"github.com/go-redis/redis"
 	"github.com/antchfx/xmlquery"
 	"github.com/antchfx/jsonquery"
 	"github.com/Azure/sonic-mgmt-common/cvl/internal/yparser"
 	. "github.com/Azure/sonic-mgmt-common/cvl/internal/util"
 	"sync"
 	"flag"
-	"runtime"
+	//"runtime"
```
Comment (renukamanavalan) [inline cvl/cvl.go:36]: Same. Please remove

File: cvl/cvl_cache.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+import (
+	"encoding/json"
+	"github.com/go-redis/redis"
+	//lint:ignore ST1001 This is safe to dot import for util package
+	. "github.com/Azure/sonic-mgmt-common/cvl/internal/util"
+	"github.com/Azure/sonic-mgmt-common/cvl/internal/yparser"
+	"time"
+	"runtime"
+	"github.com/antchfx/jsonquery"
+	//"github.com/antchfx/xmlquery"
```
Comment (renukamanavalan) [inline cvl/cvl_cache.go:30]: Same. Please remove

File: cvl/cvl_semantics.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	//lint:ignore ST1001 This is safe to dot import for util package
+	. "github.com/Azure/sonic-mgmt-common/cvl/internal/util"
+)
+
+//YValidator YANG Validator used for external semantic
+//validation including custom/platform validation
+type YValidator struct {
+	root *xmlquery.Node    //Top evel root for data
+	current *xmlquery.Node //Current position
+	//operation string     //Edit operation
```
Comment (renukamanavalan) [inline cvl/cvl_semantics.go:35]: Same. Please remove

File: cvl/cvl_test.go [hunk 5]
```diff
@@ -32,7 +32,7 @@ import (
 	"testing"
 	"runtime"
 	. "github.com/Azure/sonic-mgmt-common/cvl/internal/util"
-	"github.com/Azure/sonic-mgmt-common/cvl/internal/yparser"
+//	"github.com/Azure/sonic-mgmt-common/cvl/internal/yparser"
```
Comment (renukamanavalan) [inline cvl/cvl_test.go:35]: Same. Please remove

File: cvl/cvl_test.go [hunk 6]
```diff
@@ -2861,6 +2861,7 @@ func TestValidateEditConfig_Delete_Entry_Then_Dep_Leafref_Positive(t *testing.T)
 	unloadConfigDB(rclient, depDataMap)
 }
 
+/*
 func TestBadSchema(t *testing.T) {
```
Comment (renukamanavalan) [inline cvl/cvl_test.go:2865]: Same. Please remove

Comment (dutta-partha) [reply cvl/cvl_test.go]: Done.

File: cvl/cvl.go (hunk 1 above)
Comment (dutta-partha) [reply cvl/cvl.go]: Done.

File: cvl/cvl.go (hunk 2 above)
Comment (dutta-partha) [reply cvl/cvl.go]: Done

File: cvl/cvl_cache.go (hunk 3 above)
Comment (dutta-partha) [reply cvl/cvl_cache.go]: Done

File: cvl/cvl_semantics.go (hunk 4 above)
Comment (dutta-partha) [reply cvl/cvl_semantics.go]: Done

File: cvl/cvl_test.go (hunk 5 above)
Comment (dutta-partha) [reply cvl/cvl_test.go]: Done.

**Pull request merged**